### ISSUE TITLE
`unchecked_time_subtraction` misses `Ops::sub` method call

### ISSUE THREAD

**Issue by vilgotf** [rust-lang/rust-clippy#16230]
### Summary

The `unchecked_time_subtraction` lint does not catch `Ops::sub` usage.

### Lint Name

unchecked_time_subtraction

### Reproducer

I tried this code:

```rust
Duration::ZERO.sub(Duration::MAX);
```

I expected to see this happen: clippy to report a warning

Instead, this happened: clippy did not report a warning

### Version

```text
rustc 1.92.0 (ded5c06cf 2025-12-08)
binary: rustc
commit-hash: ded5c06cf21d2b93bffd5d884aa6e96934ee4234
commit-date: 2025-12-08
host: x86_64-pc-windows-msvc
release: 1.92.0
LLVM version: 21.1.3
```
